File: 251004/Kiseliov/lab5/TaskRedis/Module.Discussion/Mappers/NoticeMapper.cs
using Discussion.Dto.Request;
using Discussion.Dto.Response;
using Discussion.Models;
using Riok.Mapperly.Abstractions;
namespace Discussion.Mappers;

[Mapper]
public static partial class NoticeMapper
{
    public static partial Notice ToEntity(this NoticeRequestTo noticeRequestTo);
    public static partial NoticeResponseTo ToResponse(this Notice notice);
    public static partial IEnumerable<NoticeResponseTo> ToResponse(this IEnumerable<Notice> notices);

    public static Notice UpdateEntity(this Notice notice, NoticeRequestTo updateNoticeRequestTo) => new()
    {
        Id = notice.Id,
        TweetId = updateNoticeRequestTo.TweetId,
        Content = updateNoticeRequestTo.Content,
        Country = notice.Country
    };
}

File: 251003/Stasevichy/discussion/src/main/java/com/lab/labDC/mapper/NoticeMapper.java
package com.lab.labDC.mapper;

import com.lab.labDC.dto.NoticeRequestTo;
import com.lab.labDC.dto.NoticeResponseTo;
import com.lab.labDC.entity.Notice;
import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

@Mapper
public interface NoticeMapper {

    NoticeMapper INSTANCE = Mappers.getMapper(NoticeMapper.class);

    Notice toEntity(NoticeRequestTo noticeRequestTo);

    NoticeResponseTo toResponseDto(Notice notices);

    Notice toEntity(NoticeResponseTo noticeResponseTo);

    NoticeRequestTo toRequestDto(Notice notices);
}
